src/features/resume-review.ts

```diff
@@ -25,34 +36,55 @@ const sendResumeMessage = async (msg: Message): Promise<true | void> => {
   );
   if (!builder) {
     logger.log("[RESUME]", "Failed to generate resume PDF in thread");
-    return;
+    return [];
   }
 
-  await msg.channel.send({
-    files: builder,
-  });
-
-  return true;
+  return builder;
 };
 
 const resumeReviewPdf: ChannelHandlers = {
-  handleMessage: async ({ msg }) => {
-    // NOTE: This cast is safe as we are fetching the actual message in the index.ts/handleMessage
-    const message = msg as Message;
-    const cooldownKey = `resume-${msg.channelId}`;
+  handleMessage: async ({ msg: message }) => {
+    if (
+      !message.channel.isThread() ||
+      message.channel.parentId !== CHANNELS.resumeReview
+    ) {
+      return;
+    }
+    const cooldownKey = `resume-${message.channelId}`;
 
     if (cooldown.hasCooldown(message.author.id, cooldownKey)) {
+      message.channel.send(
+        "You posted < 5 minutes ago, not creating a new preview",
+      );
       return;
     }
 
-    const sucess = await sendResumeMessage(message);
-    if (sucess) {
+    await message.channel.sendTyping();
+
+    const images = await buildResumeImages(message);
+    if (images.length > 0) {
```

Comment:
```suggestion
    if (images.length <= 0) return;
```

We should do early-return here to not send embed to the user without any images.

Reply:
Oh word yeah. This suggestion is gonna bork the close brace but good point

Reply:
See https://github.com/reactiflux/reactibot/pull/375/files/211b595e840bbf8c9c363ceafb6bf5d1632174fd#r1629821110